class/practice2.2/main.cpp

```diff
@@ -0,0 +1,204 @@
+#include <string>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <ranges>
+#include <print>
+#include <cstdint>
+
+namespace {
+   constexpr const int MatrixSize = 16;
+}
+
+std::vector<std::string> splitString(const std::string& str, const char delimiter) {
+    return str  | std::ranges::views::split(delimiter)
+       | std::ranges::to<std::vector<std::string>>();
+}
+
+struct Color {
+    int r{ -1 };
+    int g{ -1 };
+    int b{ -1 };
+    };
+
+class Pixel{
+public:
+    Pixel() = default;
+    Pixel(const std::string& col){
+       auto values = splitString(col, ',');
+       if (values.size() == 3) {
+        color.r=convertToInt(values[0]);
```

Comment:
minor some bad indentation here - it is better to have consistent code formatting to simplify reading of the code.